vertx-web/src/main/java/io/vertx/ext/web/handler/sockjs/impl/BaseTransport.java

```diff
@@ -156,7 +156,11 @@ static void setJSESSIONID(SockJSHandlerOptions options, RoutingContext rc) {
     }
   }
 
-  static void setCORS(RoutingContext rc) {
+  static void setCORS(SockJSHandlerOptions options, RoutingContext rc) {
+    if (!options.isCorsHeadersEnabled()) {
```

Comment:
I believe we could also check if the cors headers are already present e.g.: cors handler is set up in route, perhaps this would be enough to not need an options object here?